zipline/pipeline/loaders/quarter_estimates.py

```diff
@@ -0,0 +1,287 @@
+import pandas as pd
+from six import viewvalues
+from toolz import groupby
+from zipline.pipeline.common import (
+    EVENT_DATE_FIELD_NAME,
+    FISCAL_QUARTER_FIELD_NAME,
+    FISCAL_YEAR_FIELD_NAME,
+    SID_FIELD_NAME,
+    TS_FIELD_NAME,
+)
+from zipline.pipeline.loaders.base import PipelineLoader
+from zipline.pipeline.loaders.frame import DataFrameLoader
+
+ALL_DATES = 'dates'
+FISCAL_QUARTER = 'fiscal_quarter'
+FISCAL_YEAR = 'fiscal_year'
+NEXT_FISCAL_QUARTER = 'next_fiscal_quarter'
+NEXT_FISCAL_YEAR = 'next_fiscal_year'
+PREVIOUS_FISCAL_QUARTER = 'previous_fiscal_quarter'
+PREVIOUS_FISCAL_YEAR = 'previous_fiscal_year'
+SIMULTATION_DATES = 'dates'
+
+
+def calc_forward_shift(yrs, qtrs, num_qtrs_shift):
+    """
+    Calculate the new years and quarters based on on shifting the specified
+    number of quarters forward.
+
+    Parameters
+    ----------
+    yrs : np.Series
+        The starting years.
+    qtrs : np.Series
+        The starting quarters.
+    num_qtrs_shift : int
+        The number of quarters to shift forward.
+
+
+    Returns
+    -------
+    result_years : pd.Series
+        A series contains the new years.
+    result_qtrs : pd.Series
+        A series that contains the new quarters.
+    """
+
+    if num_qtrs_shift < 0:
+        raise AssertionError("Must pass a number of quarters >= 0")
```

Comment:
this should be a `ValueError` since it is an exception based on user input to a function